src/main/java/de/tum/cit/aet/artemis/lecture/service/SlideUnhideScheduleService.java

```diff
@@ -0,0 +1,109 @@
+package de.tum.cit.aet.artemis.lecture.service;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE_AND_SCHEDULING;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledFuture;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Service;
+
+import de.tum.cit.aet.artemis.lecture.domain.Slide;
+import de.tum.cit.aet.artemis.lecture.repository.SlideRepository;
+
+/**
+ * Scheduler implementation that is only active on nodes with the CORE_AND_SCHEDULING profile.
+ * This handles the actual scheduling of tasks.
+ */
+@Profile(PROFILE_CORE_AND_SCHEDULING)
+@Service
+public class SlideUnhideScheduleService implements ApplicationListener<ApplicationReadyEvent> {
+
+    private final SlideRepository slideRepository;
+
+    private final SlideUnhideService slideUnhideService;
+
+    private final TaskScheduler taskScheduler;
+
+    private final Map<Long, ScheduledFuture<?>> scheduledTasks = new ConcurrentHashMap<>();
+
+    private static final Logger log = LoggerFactory.getLogger(SlideUnhideScheduleService.class);
+
+    public SlideUnhideScheduleService(SlideRepository slideRepository, SlideUnhideService slideUnhideService, TaskScheduler taskScheduler) {
+        this.slideRepository = slideRepository;
+        this.slideUnhideService = slideUnhideService;
+        this.taskScheduler = taskScheduler;
+    }
+
+    /**
+     * Method called when the application is ready.
+     * It loads all hidden slides and schedules tasks to unhide them at their expiration time.
+     */
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent event) {
+        scheduleAllHiddenSlides();
+    }
+
+    /**
+     * Loads all slides with a non-null hidden timestamp and schedules their unhiding.
+     */
+    public void scheduleAllHiddenSlides() {
+        List<Slide> hiddenSlides = slideRepository.findAllByHiddenNotNull();
```

Comment:
do you really need the whole slide entity here including the ManyToOne relationships that are fetched eagerly? I don't think so. If you only need the slide id and the hidden value, I suggest you only retrieve those to make the data exchange between database and server as minimal as possible. You can use a DTO for this